bitgo-java-api/src/main/java/com/bitso/bitgo/v2/BitGoClientImpl.java

```diff
@@ -151,15 +151,15 @@ public BitGoClientImpl(String token, String baseUrl, boolean unsafe) {
                     }
                 } else if (optionalKey.equals("fee")) {
                     if (optionalValue != null) {
-                        if (optionalValue instanceof BigDecimal) {
-                            data.put(optionalKey, ((BigDecimal) optionalValue).toBigInteger());
+                        if (optionalValue instanceof Number) {
+                            data.put(optionalKey, new BigDecimal(optionalValue.toString()).toBigInteger());
                         } else {
                             throw new IllegalArgumentException("Fee should be a BigDecimal value");
                         }
                     }
                 } else if (optionalKey.equals("feeTxConfirmTarget")) {
                     if (optionalValue != null) {
-                        if (optionalValue instanceof BigDecimal) {
+                        if (optionalValue instanceof Number) {
```

Comment:
Should this data.put() be the same as the one above?

`       data.put(optionalKey, new BigDecimal(optionalValue.toString()).toBigInteger());`

Reply:
I actually was thinking that I might be good idea, I left it like that because it was like this from before, but it might make sense, everything is in satoshis anyway and the actual addressess-amounts are in biginteger, I'll change it